k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -77,36 +88,23 @@ void PodioDataSvc::endOfRead() {
   // todo: figure out sthg to do with sc (added to silence -Wunused-result)
 }
 
-void PodioDataSvc::setCollectionIDs(podio::CollectionIDTable* collectionIds) {
-  if (m_collectionIDs != nullptr) {
-    delete m_collectionIDs;
-  }
-  m_collectionIDs = collectionIds;
-}
-
 /// Standard Constructor
 PodioDataSvc::PodioDataSvc(const std::string& name, ISvcLocator* svc)
-    : DataSvc(name, svc), m_collectionIDs(new podio::CollectionIDTable()) {
-  m_eventDataTree = new TTree("events", "Events tree");
+    : DataSvc(name, svc) {
 }
 
 /// Standard Destructor
 PodioDataSvc::~PodioDataSvc() {}
 
-StatusCode PodioDataSvc::readCollection(const std::string& collName, int collectionID) {
-  podio::CollectionBase* collection(nullptr);
-
-  auto idTable = m_provider.getCollectionIDTable();
-  m_provider.get(collectionID, collection);
+StatusCode PodioDataSvc::readCollection(const std::string& collName) {
+  const podio::CollectionBase* collection(nullptr);
 
+  collection = m_frame.get(collName);
   if (collection->isSubsetCollection()) {
     return StatusCode::SUCCESS;
   }
   auto wrapper = new DataWrapper<podio::CollectionBase>;
-  int  id      = m_collectionIDs->add(collName);
-  collection->setID(id);
-  collection->prepareAfterRead();
-  wrapper->setData(collection);
+    wrapper->setData(collection);
```

Comment:
This seems to be a somewhat historical question at this point, but why are subset collections handled differently here than normal collections? It looks to me as if they would never end up in the underlying Frame by returning early here, which would mean they are not available for writing.

Reply:
Thanks for spotting the bug. Previously that was to prevent a sort of double counting in the writing part.